circuits/src/zk_circuits/valid_match_settle/single_prover.rs

```diff
@@ -333,6 +386,24 @@ where
         Ok(())
     }
 
+    /// Validate that the receive balance amounts after update do not overflow
+    fn validate_no_receive_overflow(
```

Comment:
If we are constraining the bit-length of the wallet's `match_fee` in `VALID WALLET CREATE`, do we still need these bound checks here? Same goes for the protocol fee balance, we sort of already assume it's set correctly by the contract (though that should be documented if not constrained)

Reply:
Yes we do, the balance fields can still overflow. Suppose you _started out_ with a very large balance and basically any match would put you over (there is a test for this), then regardless of the validation we've done to the fee, we still will violate the invariants. We really should strive to hold these as invariants (absolute) rather than rules of thumb, I'll give two reasons why:
1. The invariants around wallet value ranges are mostly to avoid overflowing/underflowing the field, but they also simplify the circuit designs of other circuits. If we maintain them absolutely we don't have to worry as much about interconnectedness.
2. Fewer total range checks need to be done in the system if we assume that each circuit does its part to maintain the invariants with no exeception